### PR TITLE
Adopt Gradle 8

### LINKED ISSUE #321: The `Provider.forUseAtConfigurationTime` method has been deprecated.
**Description**

> [Provider#forUseAtConfigurationTime](https://docs.gradle.org/8.3/javadoc/org/gradle/api/provider/Provider.html#forUseAtConfigurationTime--) is now deprecated and scheduled for removal in Gradle 9.0. Clients should simply remove the call.

https://docs.gradle.org/8.3/userguide/upgrading_version_7.html?#for_use_at_configuration_time_deprecation

**Versions**

- Test logger version: 3.2.0
- Gradle version: 8.3
- Java version: 17

### PR DESCRIPTION
**Pull request by Goooler**
Closes #321.

### CODE REVIEW

**Comment by Goooler** [comment]
You can rebase these commits and replace @dependabot with @renovate-bot.

**Comment by radarsh** [comment]
@Goooler thanks for this pull request. Can you justify why are you switching to renovate bot? I much prefer to use dependabot as it is built into GitHub now.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```
Comment (radarsh) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can you use the `all` distribution please?

Comment (Goooler) [reply gradle/wrapper/gradle-wrapper.properties:3]: Actually, Gradle sources could be attached to projects by IDEA for now. I can still use `all` if you want.

**Comment by Goooler** [comment]
> Can you justify why are you switching to renovate bot? I much prefer to use dependabot as it is built into GitHub now.

@renovate-bot can send the Gradle updates, and supports more kinds of dep updates.

File: build.gradle [hunk 2]
```diff
@@ -44,14 +44,18 @@ dependencies {
...
 
     functionalTestImplementation localGroovy()
     functionalTestImplementation gradleTestKit()
     functionalTestImplementation 'org.spockframework:spock-core:2.0-groovy-3.0'
-    functionalTestImplementation 'commons-io:commons-io:2.11.0'
+    functionalTestImplementation 'commons-io:commons-io:2.14.0'
+}
+
+java {
+    toolchain.languageVersion = JavaLanguageVersion.of(8)
```
Comment (radarsh) [inline build.gradle:58]: Let's make it 17 and I will release a major version.

Comment (Goooler) [reply build.gradle:58]: If we use 17 here, it requires all consumers running on Java 17 above, we should keep the compatibility of Java 8.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (radarsh) [reply gradle/wrapper/gradle-wrapper.properties:3]: Good to know. Leave it using `bin` for now and if something doesn't work we can always restore `all` without having to release a new version

File: build.gradle (hunk 2 above)
Comment (Goooler) [reply build.gradle:58]: Pin 8 here for compatibility, use Java 17 on CI at 8ff89d42fb64e50ff20231fc72ede3d19ee8bdc7.

**Comment by radarsh** [comment]
Thanks. Lastly, is this PR going to be compatible with Gradle 7?

**Comment by Goooler** [comment]
> is this PR going to be compatible with Gradle 7?

It should work, maybe we should add more Gradle versions into test matrix.

**Comment by Goooler** [comment]
Did a quick test at https://github.com/Goooler/gradle-test-logger-plugin/pull/27, seems need to fix some tests.

**Comment by radarsh** [comment]
No problem, looks like [some classes introduced in 7.6](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/testing/TestFailure.html) are being used in your branch indirectly. I'll do some more checks later and merge this. Thanks once again.

**Comment by radarsh** [comment]
Thanks once again for a great PR

**Pull request merged**